theming.module

```diff
@@ -31,3 +31,35 @@ function theming_theme_registry_alter(&$hooks) {
     $hook['preprocess functions'] = array_unique($hook['preprocess functions']);
   }
 }
+
+/**
+ * Implements hook_element_info_alter().
+ */
+function theming_element_info_alter(&$info) {
+  $info['scripts']['#pre_render'] = ['theming_pre_render_scripts'];
+}
+
+/**
+ * Wrap drupal_pre_render_scripts() to add custom attributes to script tags.
+ */
+function theming_pre_render_scripts(array $elements) {
+  $rendered_elements = drupal_pre_render_scripts($elements);
+  foreach ($elements['#items'] as $key => $script) {
+    // Local and external scripts keep their path/URL as key, inline scripts and settings don’t.
+    if (is_numeric($key) || $key === 'settings') {
+      continue;
+    }
+    $custom_attributes = $script['attributes'] ?? [];
+    foreach (element_children($rendered_elements['scripts']) as $index) {
```

Comment:
Wouldn't it be easier to make this the main loop and look for `$elements['#items'][$attributes['src']]['attributes']`? :thinking:

Reply:
True. I’ve turned it aroud in a new commit.